Add ability to specify serviceaccount for cribl leader

bpeynetti-stytch opened the pull request:
## Summary

Enables attaching a service account to the leader pod

## Why this is necessary

As https://github.com/criblio/helm-charts/issues/212 explains, if we want to use AWS KMS as the key management mechanism instead of something like the Cribl stream KMS, we need to also provide the leader permissions related to AWS IAM in order to interact with the AWS KMS service. That can be possible via hard-coding AWS access keys, or by using the IAM role attached to the instance the pod is scheduled on. This adds the ability to provide attach a service account to the leader pod similar to how it is already possible with the workers.  

## Tests

```
$ helm unittest logstream-leader

### Chart [ logstream-leader ] logstream-leader

 PASS  Persistent Volume Claims logstream-leader/tests/claims_test.yaml
 PASS  ConfigMap        logstream-leader/tests/configmap_test.yaml
 PASS  Deployment       logstream-leader/tests/deployment_test.yaml
 PASS  Extras   logstream-leader/tests/extras_test.yaml
 PASS  Ingress  logstream-leader/tests/ingress_test.yaml
 PASS  Labels   logstream-leader/tests/labels_test.yaml
 PASS  Deployment Security Context      logstream-leader/tests/securitycontext_test.yaml
 PASS  Service  logstream-leader/tests/service_test.yaml
 PASS  Service Account  logstream-leader/tests/serviceaccount_test.yaml

Charts:      1 passed, 1 total
Test Suites: 9 passed, 9 total
Tests:       53 passed, 53 total
Snapshot:    0 passed, 0 total
Time:        103.195459ms
```

pdugas commented:
In the chart for Edge, we handle when `create` is set and `name` isn't by defaulting the account name to the fullname template. Seems like a corner case we should handle similarly here.

bpeynetti-stytch commented:
Thanks for the review @pdugas! Made a few tweaks:
- Adds annotations similar to how it works in Edge, and added them to the README for the leader. Also added a new unit test for that
- The behavior you specified about defaulting the account name to the fullname was already implemented, but I added that to the documentation and comments. The unit tests already cover this use case

sdball reviewed (approved):
This looks great! But should we call out in the README that `serviceAccount.create` will create a service account with no permissions that would then need to be configured?
